frontend/src/lib/v2/WorkflowUtils.ts

```diff
@@ -48,7 +49,9 @@ export function isTemplateV2(templateString: string): boolean {
 
 // Assuming template is the JSON format of PipelineSpec in api/v2alpha1/pipeline_spec.proto
 export function convertYamlToV2PipelineSpec(template: string): PipelineSpec {
-  const pipelineSpecYAML = jsyaml.safeLoad(template);
+  // If pipeline_spec exists in the return value of safeload, which means the original yaml contains
+  // platform_spec, then the pipeline spec is stored in pipeline_spec field.
+  const pipelineSpecYAML = jsyaml.safeLoad(template)['pipeline_spec'] ?? jsyaml.safeLoad(template);
```

Comment:
'pipeline_spec' is a constant which is defined by the IR. Recommend to define a global variable for `pipeline_spec` to be used consistently across the FE.